velox/exec/HashBuild.cpp

```diff
@@ -468,9 +468,10 @@ bool HashBuild::reserveMemory(const RowVectorPtr& input) {
   if (numRows != 0) {
     // Test-only spill path.
     if (testingTriggerSpill()) {
-      numSpillRows_ = std::max<int64_t>(1, numRows / 10);
-      numSpillBytes_ = numSpillRows_ * outOfLineBytesPerRow;
-      return false;
+      memory::testingRunArbitration();
+      // NOTE: the memory arbitration should have triggered spilling on this
+      // hash build operator so we return true to indicate have enough memory.
+      return true;
```

Comment:
Operator::ReclaimableSectionGuard guard(this);
      SuspendedSection suspendedSection(operatorCtx_->driver());
      memory::testingRunArbitration();
      return true;

Reply:
Yeah, I use testingRunArbitration(pool) which will take care of that. Thanks!

Reply:
I guess you also will need the reclaimable section guard?
Operator::ReclaimableSectionGuard guard(this);